api/db/repository/petition.py

```diff
@@ -62,12 +76,90 @@ def get_by_id(_db: Session, petition_id: int, petition_status=""):
     return db_petition.first()
 
 
-def get_all(_db: Session, offeset: int = 0, limit: int = 100):
-    return _db.query(Petition).offset(offeset).limit(limit).all()
+def get_all(
+    _db: Session,
+    offeset: int = 0,
+    limit: int = 100,
+    search_query: str = "",
+    filtering: FilteringRequest = None,
+    ordering: List[OrderingRequest] = None,
+):
+    query = _db.query(Petition).filter(
+        Petition.title.contains(search_query)
+    )
+    if filtering:
+        for key, value in filtering:
+            if value and key in ALLOWED_FILTERS:
+                if key == "status":
+                    query = query.filter(Petition.status.in_(value))
+                else:
+                    query = query.filter(
+                        and_(
+                            value.start != DEFAULT_DATE,
+                            getattr(Petition, key) >= value.start,
+                        )
+                    ).filter(
+                        and_(
+                            value.end != DEFAULT_DATE,
+                            getattr(Petition, key) <= value.end,
+                        )
+                    )
+            else:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Can not apply such filtering",
+                )
+
+    if ordering:
+        for order in ordering:
```

Comment:
You should order_by only one time,  cause you'll get result of the last ordering only. If you want to do ordering by multiple columns they should be in one statement https://stackoverflow.com/questions/15791760/how-can-i-do-multiple-order-by-in-flask-sqlalchemy (but I think one column is enough for now)

Reply:
I fixed it